### ISSUE TITLE
[FR] Argmax each batch in predict_classes

### ISSUE THREAD

**Issue by ohadle** [fchollet/keras#5081]
Currently, [`predict_classes`](https://github.com/fchollet/keras/blob/master/keras/models.py#L834) first calculates probabilities for all the data and then performs the `argmax`.  For large datasets, with many softmax bins, this is not memory-efficient.  
Instead, keras could perform the `argmax` [per-batch](https://github.com/fchollet/keras/blob/master/keras/engine/training.py#L907).
